From 0bb016ec6ebf84735304c5b35d8972b90fb79af3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 13 Oct 2025 23:14:14 +0200
Subject: [PATCH] Ensure the window is within the screen even if positioned

Some clients like Qt 6.10 sends a static position to the top left corner,
which needs to be moved to fit the window frame to the screen.
---
 openbox/client.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/openbox/client.c b/openbox/client.c
index 6e53eb3c3..b8c8778e1 100644
--- a/openbox/client.c
+++ b/openbox/client.c
@@ -412,8 +412,7 @@ void client_manage(Window window, ObPrompt *prompt)
                              ob_state() == OB_STATE_RUNNING &&
                              (self->type == OB_CLIENT_TYPE_DIALOG ||
                               self->type == OB_CLIENT_TYPE_SPLASH ||
-                              (!((self->positioned & USPosition) ||
-                                 settings->pos_given) &&
+                              (!settings->pos_given &&
                                client_normal(self) &&
                                !self->session &&
                                /* don't move oldschool fullscreen windows to
